designer_v2/lib/features/monitor/study_monitor_page.dart

```diff
@@ -1,20 +1,242 @@
 
 class StudyMonitorScreen extends StudyPageWidget {
   const StudyMonitorScreen(super.studyId, {super.key});
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    return Container(
-      width: double.infinity,
-      color: Theme.of(context).colorScheme.secondary.withOpacity(0.03),
-      height: 300,
-      child: const Center(
-        child: UnderConstruction(),
+    final state = ref.watch(studyControllerProvider(studyId));
+    return AsyncValueWidget<Study>(
+      value: state.study,
+      data: (study) {
+        final studyMonitorData = study.monitorData;
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            _monitorSectionHeader(context, studyMonitorData),
+            const SizedBox(height: 40.0), // spacing between body elements
+            if (studyMonitorData.items.isNotEmpty)
+              StudyMonitorTable(
+                ref: ref,
+                studyMonitorItems: studyMonitorData.items,
+                onSelectItem: (item) =>
+                    _onSelectParticipant(context, ref, item, study),
+              )
+            else
+              EmptyBody(
+                icon: Icons.person_off_rounded,
+                title: tr.monitoring_no_participants_title,
+                description: tr.monitoring_no_participants_description,
+              ),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _monitorSectionHeader(
+    BuildContext context,
+    StudyMonitorData monitorData,
+  ) {
+    final int total = monitorData.items.length;
+    const double minPercentage = 0.05; // Minimum percentage for visibility
+
+    double activePercentage = monitorData.activeParticipants / total;
+    double inactivePercentage = monitorData.inactiveParticipants / total;
+    double dropoutPercentage = monitorData.dropoutParticipants / total;
+    double completedPercentage = monitorData.completedParticipants / total;
+
+    // Adjust for minimum percentage visibility
+    if (monitorData.activeParticipants == 0) activePercentage = minPercentage;
+    if (monitorData.inactiveParticipants == 0) {
+      inactivePercentage = minPercentage;
+    }
+    if (monitorData.dropoutParticipants == 0) dropoutPercentage = minPercentage;
+    if (monitorData.completedParticipants == 0) {
+      completedPercentage = minPercentage;
+    }
+
+    // Normalize the percentages so they add up to 1.0
+    final double sumOfPercentages = activePercentage +
+        inactivePercentage +
+        dropoutPercentage +
+        completedPercentage;
+
+    activePercentage /= sumOfPercentages;
+    inactivePercentage /= sumOfPercentages;
+    dropoutPercentage /= sumOfPercentages;
+    completedPercentage /= sumOfPercentages;
+
+    const Color activeColor = Color(0xFF0072B2);
+    const Color inactiveColor = Color(0xFFD55E00);
+    const Color dropoutColor = Color(0xFFCC79A7);
+    const Color completedColor = Color(0xFF009E73);
```

Comment:
How about using the dark purple color (I think it's the primary color) of the monitoring table for the `activeColor` to keep it consistent? The blue color could instead be used for dropout for example, to avoid similarity.

```suggestion
    const Color activeColor = theme.primaryColor;
    const Color inactiveColor = Color(0xFFD55E00);
    const Color dropoutColor = Color(0xFF0072B2);
    const Color completedColor = Color(0xFF009E73);
```

Reply:
yeah. thats good idea to keep the design consistant.